doc/specs/#5000 - Process Model 2.0/#4472 - Windows Terminal Session Management.md

```diff
@@ -0,0 +1,560 @@
+command:
+
+#### `--window,-w window-id`
+Run these commands in the given Windows Terminal session. This enables opening
+new tabs, splits, etc. in already running Windows Terminal windows.
+* If `window-id` is `0`, run the given commands in _the current window_.
+* If `window-id` is a negative number, or the reserved name `new`, run the
+  commands in a _new_ Terminal window.
+* If `window-id` is the ID or name of an existing window, then run the
+  commandline in that window.
+* If `window-id` is _not_ the ID or name of an existing window, create a new
+  window. That window will be assigned the ID or name provided in the
+  commandline. The provided subcommands will be run in that new window.
+* If `window-id` is omitted, then obey the value of `windowingBehavior` when
+  determining which window to run the command in.
+
+_Whenever_ `wt.exe` is started, it must _always_ pass the provided commandline
+first to the monarch process for handling. This is important for glomming
+scenarios (as noted above). The monarch will parse the commandline, determine
+which window the commandline is destined for, then call `ExecuteCommandline` on
+that peasant, who will then run the command.
+
+#### Running commands in the current window:`wt --window 0`
+
+If `wt -w 0 <commands>` is run _outside_ a WT instance, it could attempt to glom
+onto _the most recent WT window_ instead. This seems more logical than something
+like `wt --window last` or some other special value indicating "run this in the
+MRU window".<sup>[[2]](#footnote-2)</sup>
+
+That might be a simple, but **wrong**, implementation for "the current window".
+If the peasants always raise an event when their window is focused, and the
+monarch keeps track of the MRU order for peasants, then one could naively assume
+that the execution of `wt -w 0 <commands>` would always return the window the
+user was typing in, the current one. However, if someone were to do something
+like `sleep 10 ; wt -w 0 <commands>`, then the user could easily focus another
+WT window during the sleep, which would cause the MRU window to not be the same
+as the window executing the command.
+
+To solve this issue, we'll  other than
+attempting to use the `WT_SESSION` environment variable. If a `wt.exe` process
+is spawned and that's in it's environment variables, it could try and ask the
+monarch for the peasant who's hosting the session corresponding to that GUID.
+This is more of a theoretical solution than anything else.
+
+In the past we've been reluctant to rely too heavily on `WT_SESSION`. However,
+an environment variable does seem to be the only reliable way to be confident
+where the window was created from. We could introduce another environment
+variable instead - `WT_WINDOW_ID`. That would allow us to shortcut the session
+ID lookup. However, I worry about exposing the window ID as an environment
+variable. If we do that, users will inevetably use that instead of the `wt -0`

```

Comment:
Also, we **would not** be able to change `WT_WINDOW_ID` when you dragged a tab around (!)